app/javascript/profileDropdown/flagButton.js

```diff
@@ -15,53 +17,56 @@ export function initFlag() {
     return;
   }
 
-  const user = userData();
-  if (!user) {
-    return;
-  }
+  getUserDataAndCsrfToken().then(({ _current_user, _csrfToken }) => {
+    const user = userData();
+    if (!user) {
+      flagButton.remove;
+      return;
+    }
 
-  const { profileUserId, profileUserName } = flagButton.dataset;
-  let isUserFlagged = flagButton.dataset.isUserFlagged === 'true';
-  const trustedOrAdmin = user.trusted || user.admin;
+    const { profileUserId, profileUserName } = flagButton.dataset;
+    let isUserFlagged = flagButton.dataset.isUserFlagged === 'true';
```

Comment:
hmm - eslint let this slip through. I see we _do_ mutate it during the button action (if the post succeeded then the state inverts, since it's a toggle):

```javascript
          if (response.result === 'create') {
            isUserFlagged = true;
            flagButton.innerHTML = `Unflag ${userName}`;
          } else {
            isUserFlagged = false;
```